opentelemetry-sdk/src/metrics/periodic_reader_with_own_thread.rs

```diff
@@ -0,0 +1,798 @@
+                                // Reset the interval if the flush finishes after the expected export time
+                                // effectively missing the normal export.
+                                // Should we attempt to do the missed export immediately?
+                                // Or do the next export at the next interval?
+                                // Currently this attempts the next export immediately.
+                                // i.e calling Flush can affect the regularity.
+                                interval_start = Instant::now();
+                                remaining_interval = Duration::ZERO;
+                            }
+                        }
+                        Ok(Message::Shutdown(response_sender)) => {
+                            // Perform final export and break out of loop and exit the thread
+                            otel_debug!(name: "PeriodReaderThreadExportingDueToShutdown");
+                            if let Err(_e) = cloned_reader.collect_and_export(timeout) {
+                                response_sender.send(false).unwrap();
+                            } else {
+                                response_sender.send(true).unwrap();
+                            }
+                            break;
+                        }
+                        Err(mpsc::RecvTimeoutError::Timeout) => {
+                            let export_start = Instant::now();
+                            otel_debug!(
+                                name: "PeriodReaderThreadExportingDueToTimer",
+                                event_name = "PeriodReaderThreadExportingDueToTimer"
+                            );
+
+                            if let Err(_e) = cloned_reader.collect_and_export(timeout) {
+                                otel_debug!(
+                                    name: "PeriodReaderThreadExportingDueToTimerFailed",
+                                    event_name = "PeriodReaderThreadExportingDueToTimerFailed"
+                                );
+                            }
+
+                            let time_taken_for_export = export_start.elapsed();
+                            if time_taken_for_export > interval {
+                                otel_debug!(
+                                    name: "PeriodReaderThreadExportTookLongerThanInterval",
+                                    event_name = "PeriodReaderThreadExportTookLongerThanInterval"
+                                );
+                                // if export took longer than interval, do the
+                                // next export immediately.
+                                // Alternatively, we could skip the next export
+                                // and wait for the next interval.
+                                // Or enforce that export timeout is less than interval.
+                                // What is the desired behavior?
```

Comment:
Perhaps hybrid approach can be one option - enforce  that the export `timeout` is configured to be less than the `interval`. And then, if the delay is slightly more than the interval, we can do next export immediately. However, if the delay is in multiple of intervals, it signifies heavy load on system, and better to skip next cycle - Just a thought, but good to discuss for further PRs.